src/main/java/Duke.java

```diff
@@ -1,10 +1,342 @@
+     * @param list     The array containing all information about existing tasks and their completion status.
+     * @param listSize The current number of tasks populating the "list" array.
+     * @throws DukeException The custom defined exception relating to all errors specific to the Duke program.
+     */
+    private static void createNewEvent(String line, Task[] list, int listSize) throws DukeException {
+        int fromIndex = line.indexOf("/from");
+        int toIndex = line.indexOf("/to");
+        if (fromIndex == -1 || toIndex == -1) {
+            throw new DukeException();
+        }
+        String event;
+        String startDate;
+        String endDate;
+        if (fromIndex < toIndex) {
+            //If user typed "/from" before "/to".
+            event = line.substring(line.indexOf("event") + 5, fromIndex - 1);
+            startDate = line.substring(fromIndex + 6, toIndex - 1);
+            endDate = line.substring(toIndex + 4);
+        } else {
+            //If user typed "/to" before "/from".
+            event = line.substring(line.indexOf("event") + 5, toIndex - 1);
+            startDate = line.substring(toIndex + 4, fromIndex - 1);
+            endDate = line.substring(fromIndex + 6);
+        }
+        list[listSize] = new Event(event, startDate, endDate);
+    }
+
+    /**
+     * Prints a series of strings to inform the user that the task has been added to the list of tasks, while
+     * incrementing the "listSize" variable used to track the total tasks in the list by one.
+     *
+     * @param list     The array containing all information about existing tasks and their completion status.
+     * @param listSize The current number of tasks populating the "list" array.
+     * @return Returns the new total number of tasks found in the list.
+     */
+    private static int printNewlyAddedTask(Task[] list, int listSize) {
+        lineSeparator();
+        System.out.println("Got it. I've added this task:");
+        System.out.println("  " + list[listSize].toString());
+        listSize += 1;
+        System.out.println("Now you have " + listSize + " tasks in the list.");
+        lineSeparator();
+        return listSize;
+    }
+
+    /**
+     * Changes the completion status of the user-created task to done, and prints a series of strings
+     * to inform the user about the update to the task completion status.
+     *
+     * @param list     The array containing all information about existing tasks and their completion status.
+     * @param listSize The current number of tasks populating the "list" array.
+     * @param command  The string array containing all individual strings separated by a space (" ") character in
+     *                 the user inputted string.
+     * @throws DukeException The custom defined exception relating to all errors specific to the Duke program.
+     */
+    private static void markAsDone(Task[] list, int listSize, String[] command) throws DukeException {
```

Comment:
The `markAsDone` and `markAsUndone` functions seem pretty similar and one seems like the complement of the other. Maybe you can try extracting some functionality so as to not make it seem like you are repeating yourself.